### DISCUSSION TITLE
Jotai v2 can't set function into an atom throught Store Provider

### DISCUSSION THREAD

**Discussion by xXBlackBulletXx** [pmndrs/jotai#1746] [Show and tell] [answered]
Hi, I was trying to set a callback in an atom and use it throughout the atom, but the atom seems to be undefined. I created a CodeSandbox example, let me know if I'm missing something.
https://codesandbox.io/s/peaceful-bouman-qoxsnb

**Comment by dai-shi** [comment, marked as answer]
It's because you are not using the store you created.
I'd suggest not to use createStore, unless you are super familiar with it.

**Reply by xXBlackBulletXx** [reply]
Basically, I forgot how to save functions in Atom and I should use `() => saveFunction` for it to work as expected.
Thank you for your patience.

**Reply by dai-shi** [reply]
You shouldn't store a function directly. Wrap it in an object.
```js
const fnAtom = atom({ fn: () => "your function" })
```
